### PR TITLE
fix: fix errors in next.js

### PR DESCRIPTION
**Pull request by yunnian**
fix  two errors in next.js:
1. Part of [#3663](https://github.com/toeverything/blocksuite/issues/3663)   document and window is not defined
2. Unable to build in next.js if throw error , because next.js is prerendered on the server and hydrated on the client.

### CODE REVIEW

File: packages/blocks/src/index.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ if (env[importIdentifier] === true) {
 }
 
 if (typeof window === 'undefined') {
-  throw new Error(
+  console.error(
```
Comment (himself65) [inline packages/blocks/src/index.ts]: I think it is indeed an error, we shouldn't just ignore it.

File: packages/editor/src/index.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ if (env[importIdentifier] === true) {
 }
 
 if (typeof window === 'undefined') {
-  throw new Error(
+  console.error(
```
Comment (himself65) [inline packages/editor/src/index.ts]: NO, editor cannot be imported in SSR mode Whatever how

**Review by himself65 — CHANGES_REQUESTED** [review]

File: packages/blocks/src/index.ts (hunk 1 above)
Comment (yunnian) [reply packages/blocks/src/index.ts:63]: In the latest next.js, even with the use of client mode, it still have some server-side rendering's works, so throwing exceptions here will cause build(npm run build) errors. `npm run dev` can work, but there are also error reminders

Do you have a good way to solve the build errors  in next.js ? We can discuss it .

Blocksuite cannot be rendered on the server, I think this reminder in console is enough, We also can add a notice in the doc.  Or do you have any other good ideas？

image

Comment (himself65) [reply packages/blocks/src/index.ts]: Yeah, the error here is correct. Your code here is still not correct.

'use client' also means it has SSR.

Don't hide the error. Next.js intends to run the code in a node.js environment

Comment (yunnian) [reply packages/blocks/src/index.ts:63]: @himself65 
I don't understand Why my code is incorrect. 

In the next.js app router mode, the `use client` is used to represent the client : https://nextjs.org/docs/app/building-your-application/rendering/client-components

How to write the correct code   in next.js's app router mode ？

Screenshot 2023-09-19 at 02 22 59

Comment (himself65) [reply packages/blocks/src/index.ts]: https://nextjs.org/docs/app/building-your-application/optimizing/lazy-loading#skipping-ssr

**Review by doodlewind — APPROVED** [review]
Thanks for the help! Also /cc @sxzz to see if the code snippet provided here could be reused in the ecosystem-ci.

**Pull request merged**